### ISSUE TITLE
Cannot provision nodes on Amazon EC2 with RHEL 7 AMI

### ISSUE THREAD

**Issue by danbarr** [rancher/rancher#32011]
**What kind of request is this (question/bug/enhancement/feature request):**
Bug

**Steps to reproduce (least amount of steps as possible):**
1. Create an Amazon EC2 node template referencing a RHEL 7 AMI
2. Create a cluster using the node template

**Result:**
The node provisioning process results in an error and the node fails to provision:
```
[INFO] [node-controller-rancher-machine] Waiting for machine to be running, this may take a few minutes...
[INFO] [node-controller-rancher-machine] Detecting operating system of created instance...
[INFO] [node-controller-rancher-machine] Waiting for SSH to be available...
[INFO] [node-controller-rancher-machine] Detecting the provisioner...
[INFO] [node-controller-rancher-machine] Provisioning with redhat...
[INFO] [node-controller-rancher-machine] The default lines below are for a sh/bash shell, you can specify the shell you're using, with the --shell flag.
[INFO] [node-controller-rancher-machine]
[INFO] Generating and uploading node config sandbox1
[ERROR] error syncing 'c-zdjxs/m-4r4xz': handler node-controller: Error creating machine: Error running provisioning: RHEL ssh command error: command: sudo -E yum install -y curl err: exit status 255 output: Loaded plugins: amazon-id, search-disabled-repos, versionlock, requeuing
```

The above happens using the default RHEL 7 AMI. Using our custom AMI we get a little further, the curl install works and it's able to install Docker, then fails here:
```
[INFO] [node-controller-rancher-machine] Copying certs to the local machine directory...
[INFO] [node-controller-rancher-machine] Copying certs to the remote machine...
[INFO] [node-controller-rancher-machine] The default lines below are for a sh/bash shell, you can specify the shell you're using, with the --shell flag.
[INFO] [node-controller-rancher-machine] 
[INFO] Generating and uploading node config sandbox-1
[ERROR] [node] enqueing node pool c-6nwqd:np-mjb4r
[ERROR] error syncing 'c-6nwqd/m-xld52': handler node-controller: Error creating machine: Error running provisioning: RHEL ssh command error: command: printf '%!s(MISSING)' '-----BEGIN RSA PRIVATE KEY-----, requeuing
```

**Other details that may be helpful:**
Launching the same AMI manually and running the same command, "sudo -E yum install -y curl", executes successfully.

**Environment information**
- Rancher version: 2.5.7, was also an issue in 2.4.x
- Installation option (single install/HA): HA

**Cluster information**
- Cluster type: Amazon EC2 provider
- Machine type: cloud, t3a.medium (2 CPU, 4GB RAM)
- Kubernetes version: any (doesn't get this far)
- Docker version: 19.03

**Comment by danbarr** [comment]
Still relevant

**Comment by superseb** [comment]
Please provide redacted trace or debug logs where the actual steps and outputs are supplied. It might be either a timeout issue (as it works after) or something else but the logs should be able to give us that direction.

**Comment by danbarr** [comment]
Here's part of the trace-level logging from an attempted node provisioning. Debug didn't contain anything useful between the "Provisioning with redhat..." step and the error. This was using the latest base RHEL 7.9 AMI in us-gov-west-1. It's definitely starting the yum install command but seems to give up on it after 4 minutes, before it has the chance to complete. So maybe it is a timeout?

Our custom RHEL 7 AMI gets further (it successfully installs docker) but still dies while copying the certs. It successfully creates /etc/docker/ca.pem and server.pem, then hangs on creating server-key.pem (logs "About to run SSH command" for server-key.pem but never actually sends it, then errors out 4 minutes later).

The whole trace logs are pretty huge; I could redact and send it to you over Slack rather than trying to post it here?

```
2021/07/13 16:37:58 [INFO] [node-controller-rancher-machine] (rheltest1) Calling .GetSSHKeyPath 
2021/07/13 16:37:58 [TRACE] REST UPDATE apis/management.cattle.io/v3/c-vbr27/nodes/m-k9p9b 
2021/07/13 16:37:58 [INFO] [node-controller-rancher-machine] (rheltest1) Calling .GetSSHUsername 
2021/07/13 16:37:58 [TRACE] REST UPDATE apis/management.cattle.io/v3/c-vbr27/nodes/m-k9p9b 
2021/07/13 16:37:58 [INFO] [node-controller-rancher-machine] Using SSH client type: external 
2021/07/13 16:37:58 [TRACE] REST UPDATE apis/management.cattle.io/v3/c-vbr27/nodes/m-k9p9b 
2021/07/13 16:37:58 [INFO] [node-controller-rancher-machine] Using SSH private key: /management-state/node/nodes/rheltest1/machines/rheltest1/id_rsa (-rw-------) 
2021/07/13 16:37:58 [TRACE] REST UPDATE apis/management.cattle.io/v3/c-vbr27/nodes/m-k9p9b 
2021/07/13 16:37:58 [INFO] [node-controller-rancher-machine] &{[-F /dev/null -o ConnectionAttempts=3 -o ConnectTimeout=10 -o ControlMaster=no -o ControlPath=none -o LogLevel=quiet -o PasswordAuthentication=no -o ServerAliveInterval=60 -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null ec2-user@REDACTED -o IdentitiesOnly=yes -i /management-state/node/nodes/rheltest1/machines/rheltest1/id_rsa -p 22] /usr/bin/ssh <nil>} 
2021/07/13 16:37:58 [TRACE] REST UPDATE apis/management.cattle.io/v3/c-vbr27/nodes/m-k9p9b 
2021/07/13 16:37:58 [INFO] [node-controller-rancher-machine] About to run SSH command: 
2021/07/13 16:37:58 [TRACE] REST UPDATE apis/management.cattle.io/v3/c-vbr27/nodes/m-k9p9b 
2021/07/13 16:37:58 [INFO] [node-controller-rancher-machine] sudo -E yum install -y curl 
2021/07/13 16:37:58 [TRACE] REST UPDATE apis/management.cattle.io/v3/c-vbr27/nodes/m-k9p9b 
 
2021/07/13 16:42:00 [INFO] [node-controller-rancher-machine] SSH cmd err, output: exit status 255: Loaded plugins: amazon-id, search-disabled-repos 
2021/07/13 16:42:00 [TRACE] REST UPDATE apis/management.cattle.io/v3/c-vbr27/nodes/m-k9p9b 
2021/07/13 16:42:00 [INFO] [node-controller-rancher-machine] 
rhel-7-server-rhui-rh-common-rpms                        | 2.1 kB     00:00 
2021/07/13 16:42:00 [TRACE] REST UPDATE apis/management.cattle.io/v3/c-vbr27/nodes/m-k9p9b 
2021/07/13 16:42:00 [INFO] [node-controller-rancher-machine] 
rhel-7-server-rhui-rpms                                  | 2.0 kB     00:00 
2021/07/13 16:42:00 [TRACE] REST UPDATE apis/management.cattle.io/v3/c-vbr27/nodes/m-k9p9b 
2021/07/13 16:42:00 [INFO] [node-controller-rancher-machine] 
rhui-client-config-server-7                              | 2.1 kB     00:00 
2021/07/13 16:42:00 [TRACE] REST UPDATE apis/management.cattle.io/v3/c-vbr27/nodes/m-k9p9b 
2021/07/13 16:42:00 [INFO] [node-controller-rancher-machine] 
(2/9): rhel-7-server-rhui- 0%! (BADINDEX) 0.0 B/s |    0 B   --:-- ETA 

(1/9): rhel-7-server-rhui-rh-common-rpms/7Server/x86_64/gr |  124 B   00:00 
2021/07/13 16:42:00 [TRACE] REST UPDATE apis/management.cattle.io/v3/c-vbr27/nodes/m-k9p9b 
2021/07/13 16:42:00 [INFO] [node-controller-rancher-machine] 
(2/9): rhel-7-server-rhui-rh-common-rpms/7Server/x86_64/up |  34 kB   00:00 
2021/07/13 16:42:00 [TRACE] REST UPDATE apis/management.cattle.io/v3/c-vbr27/nodes/m-k9p9b 
2021/07/13 16:42:00 [INFO] [node-controller-rancher-machine] 
(3/9): rhel-7-server-rhui-rh-common-rpms/7Server/x86_64/pr |  68 kB   00:00 
2021/07/13 16:42:00 [TRACE] REST UPDATE apis/management.cattle.io/v3/c-vbr27/nodes/m-k9p9b 
2021/07/13 16:42:00 [INFO] [node-controller-rancher-machine] 
(5/9): rhel-7-server-rhui- 0%! (BADINDEX) 0.0 B/s | 103 kB   --:-- ETA 

(4/9): rhel-7-server-rhui-rpms/7Server/x86_64/group        | 772 kB   00:00 
2021/07/13 16:42:00 [TRACE] REST UPDATE apis/management.cattle.io/v3/c-vbr27/nodes/m-k9p9b 
2021/07/13 16:42:00 [INFO] [node-controller-rancher-machine] 
(5/9): rhel-7-server-rhui-rpms/7Server/x86_64/updateinfo   | 4.1 MB   00:00 
2021/07/13 16:42:00 [TRACE] REST UPDATE apis/management.cattle.io/v3/c-vbr27/nodes/m-k9p9b 
2021/07/13 16:42:00 [INFO] [node-controller-rancher-machine] 
(7/9): rhui-client-config- 8%! (BADINDEX)6.8 MB/s | 5.0 MB   00:07 ETA 

(6/9): rhui-client-config-server-7/x86_64/group            |  124 B   00:00 
2021/07/13 16:42:00 [TRACE] REST UPDATE apis/management.cattle.io/v3/c-vbr27/nodes/m-k9p9b 
2021/07/13 16:42:00 [INFO] [node-controller-rancher-machine] 
(7/9): rhui-client-config-server-7/x86_64/updateinfo       |   92 B   00:00 
2021/07/13 16:42:00 [TRACE] REST UPDATE apis/management.cattle.io/v3/c-vbr27/nodes/m-k9p9b 
2021/07/13 16:42:00 [INFO] [node-controller-rancher-machine] 
(8/9): rhui-client-config-server-7/x86_64/primary          | 2.6 kB   00:00 
2021/07/13 16:42:00 [TRACE] REST UPDATE apis/management.cattle.io/v3/c-vbr27/nodes/m-k9p9b 
2021/07/13 16:42:00 [INFO] [node-controller-rancher-machine] 
(9/9): rhel-7-server-rhui- 72%! (BADINDEX) 13 MB/s |  41 MB   00:01 ETA 

(9/9): rhel-7-server-rhui-rpms/7Server/x86_64/primary      |  52 MB   00:01 
2021/07/13 16:42:00 [TRACE] REST UPDATE apis/management.cattle.io/v3/c-vbr27/nodes/m-k9p9b 
2021/07/13 16:42:00 [INFO] [node-controller-rancher-machine] 
rhel-7-server-rhui-rh-common-rpms: [                                    ] 1/243
rhel-7-server-rhui-rh-common-rpms: [                                    ] 2/243
rhel-7-server-rhui-rh-common-rpms: [                                    ] 3/243
rhel-7-server-rhui-rh-common-rpms: [                                    ] 4/243
rhel-7-server-rhui-rh-common-rpms: [                                    ] 5/243
rhel-7-server-rhui-rh-common-rpms: [                                    ] 6/243
rhel-7-server-rhui-rh-common-rpms: [#                                   ] 7/243
rhel-7-server-rhui-rh-common-rpms: [#                                   ] 8/243
rhel-7-server-rhui-rh-common-rpms: [#                                   ] 9/243
rhel-7-server-rhui-rh-common-rpms: [#                                  ] 10/243
rhel-7-server-rhui-rh-common-rpms: [#                                  ] 11/243
rhel-7-server-rhui-rh-common-rpms: [#                                  ] 12/243
rhel-7-server-rhui-rh-common-rpms: [#                                  ] 13/243
rhel-7-server-rhui-rh-common-rpms: [##                                 ] 14/243
rhel-7-server-rhui-rh-common-rpms: [##                                 ] 15/243
rhel-7-server-rhui-rh-common-rpms: [##                                 ] 16/243
rhel-7-server-rhui-rh-common-rpms: [##                                 ] 17/243 
2021/07/13 16:42:00 [TRACE] REST UPDATE apis/management.cattle.io/v3/c-vbr27/nodes/m-k9p9b 
2021/07/13 16:42:02 [INFO] [node-controller-rancher-machine] The default lines below are for a sh/bash shell, you can specify the shell you're using, with the --shell flag. 
2021/07/13 16:42:02 [TRACE] REST UPDATE apis/management.cattle.io/v3/c-vbr27/nodes/m-k9p9b 
2021/07/13 16:42:02 [INFO] [node-controller-rancher-machine] 
2021/07/13 16:42:02 [TRACE] REST UPDATE apis/management.cattle.io/v3/c-vbr27/nodes/m-k9p9b 
2021/07/13 16:42:02 [INFO] Generating and uploading node config rheltest1 
2021/07/13 16:42:02 [DEBUG] getNodeTemplate parsed [cattle-global-nt:nt-7l2ks] to ns: [cattle-global-nt] and n: [nt-7l2ks] 
2021/07/13 16:42:02 [TRACE] REST GET apis/management.cattle.io/v3/cattle-global-nt/nodetemplates/nt-7l2ks 
2021/07/13 16:42:02 [DEBUG] Cleaning up [/opt/jail/c-vbr27/management-state/node/nodes/rheltest1] 
2021/07/13 16:42:02 [TRACE] REST UPDATE apis/management.cattle.io/v3/c-vbr27/nodes/m-k9p9b 
2021/07/13 16:42:02 [TRACE] NodeController calling handler node-controller-sync c-vbr27/m-k9p9b 
2021/07/13 16:42:02 [TRACE] REST UPDATE apis/management.cattle.io/v3/c-vbr27/nodes/m-k9p9b 
2021/07/13 16:42:02 [TRACE] NodeController calling handler rke-worker-upgrader c-vbr27/m-k9p9b 
2021/07/13 16:42:02 [ERROR] error syncing 'c-vbr27/m-k9p9b': handler node-controller: notifying bugsnag: [Error creating machine: RHEL ssh command error: command: sudo -E yum install -y curl err: exit status 255 output: Loaded plugins: amazon-id, search-disabled-repos, requeuing 
```

**Comment by danbarr** [comment]
For what it's worth, this appears to be resolved with Rancher 2.6.0.

Oddly in 2.5.9, it does *occasionally* succeed, just not consistently and often takes multiple attempts, cycling through nodes.

**Comment by danbarr** [comment]
Tested again with 2.6.2 now that the FIPS node issue is resolved, and provisioning on RHEL 7 AMIs is working as expected. I'll go ahead and close this. Thanks.
